modules/adkernelBidAdapter.js

```diff
@@ -1,310 +1,191 @@
-      return Object.keys(syncedHostZones)
-        .reduce((m, k) => {
-          syncedHostZones[k].forEach((v) => m.push([k, v]));
-          return m;
-        }, []);
-    };
-
-    function dispatchRtbRequest(host, zone, impressions, callback) {
-      let url = buildEndpointUrl(host);
-      let rtbRequest = buildRtbRequest(impressions);
-      let params = buildRequestParams(zone, rtbRequest);
-      ajax(url, (bidResp) => {
-        bidResp = bidResp === '' ? EMPTY_BID_RESPONSE : JSON.parse(bidResp);
-        utils._each(rtbRequest.imp, (imp) => {
-          let bidFound = false;
-          utils._each(bidResp.seatbid[0].bid, (bid) => {
-            /* istanbul ignore else */
-            if (!bidFound && bid.impid === imp.id) {
-              bidFound = true;
-              callback(originalBids[imp.id], imp, bid);
-            }
-          });
-          if (!bidFound) {
-            callback(originalBids[imp.id], imp);
+export const spec = {
+
+  code: 'adkernel',
+  aliases: ['headbidding'],
+  supportedMediaTypes: [VIDEO],
+  isBidRequestValid: function(bidRequest) {
+    return 'params' in bidRequest && typeof bidRequest.params.host !== 'undefined' &&
+      'zoneId' in bidRequest.params && !isNaN(Number(bidRequest.params.zoneId));
+  },
+  buildRequests: function(bidRequests) {
+    let auctionId;
+    let dispatch = bidRequests.map(buildImp)
+      .reduce((acc, curr, index) => {
+        let bidRequest = bidRequests[index];
+        let zoneId = bidRequest.params.zoneId;
+        let host = bidRequest.params.host;
+        acc[host] = acc[host] || {};
+        acc[host][zoneId] = acc[host][zoneId] || [];
+        acc[host][zoneId].push(curr);
+        auctionId = bidRequest.bidderRequestId;
+        return acc;
+      }, {});
+    let requests = [];
+    Object.keys(dispatch).forEach(host => {
+      Object.keys(dispatch[host]).forEach(zoneId => {
+        const request = buildRtbRequest(dispatch[host][zoneId], auctionId);
+        requests.push({
+          method: 'GET',
+          url: `${window.location.protocol}//${host}/rtbg`,
+          data: {
+            zone: Number(zoneId),
+            ad_type: 'rtb',
+            v: VERSION,
+            r: JSON.stringify(request)
           }
         });
-      }, params, AJAX_REQ_PARAMS);
-    }
-
-    /**
-     * Builds complete rtb bid request
-     * @param imps collection of impressions
-     */
-    function buildRtbRequest(imps) {
-      return {
-        'id': utils.getUniqueIdentifierStr(),
-        'imp': imps,
-        'site': site,
-        'at': 1,
-        'device': {
-          'ip': 'caller',
-          'ua': 'caller'
-        }
-      };
-    }
-
-    /**
-     * Build ad-network specific endpoint url
-     */
-    function buildEndpointUrl(host) {
-      return `${window.location.protocol}//${host}/rtbg`;
+      });
+    });
+    return requests;
+  },
+  interpretResponse: function(serverResponse, request) {
+    /* istanbul ignore if */
```

Comment:
Our linter won't fail on this... so you can remove it (along with the other istanbul suppressions you've got around here)